include/bluetooth/mesh/scheduler.h

```diff
@@ -0,0 +1,144 @@
+ */
+enum {
+	BT_MESH_SCHEDULER_IN_JANUARY = 1 << 0,
+	BT_MESH_SCHEDULER_IN_FEBRUARY = 1 << 1,
+	BT_MESH_SCHEDULER_IN_MARCH = 1 << 2,
+	BT_MESH_SCHEDULER_IN_APRIL = 1 << 3,
+	BT_MESH_SCHEDULER_IN_MAY = 1 << 4,
+	BT_MESH_SCHEDULER_IN_JUNE = 1 << 5,
+	BT_MESH_SCHEDULER_IN_JULY = 1 << 6,
+	BT_MESH_SCHEDULER_IN_AUGUST = 1 << 7,
+	BT_MESH_SCHEDULER_IN_SEPTEMBER = 1 << 8,
+	BT_MESH_SCHEDULER_IN_OCTOBER = 1 << 9,
+	BT_MESH_SCHEDULER_IN_NOVEMBER = 1 << 10,
+	BT_MESH_SCHEDULER_IN_DECEMBER = 1 << 11
+};
+
+/** Representation of the days of the week
+ * in which the scheduled event is enabled.
+ */
+enum {
+	BT_MESH_SCHEDULER_ON_MONDAYS = 1 << 0,
+	BT_MESH_SCHEDULER_ON_TUESDAYS = 1 << 1,
+	BT_MESH_SCHEDULER_ON_WEDNESDAYS = 1 << 2,
+	BT_MESH_SCHEDULER_ON_THURSDAYS = 1 << 3,
+	BT_MESH_SCHEDULER_ON_FRIDAYS = 1 << 4,
+	BT_MESH_SCHEDULER_ON_SATURDAYS = 1 << 5,
+	BT_MESH_SCHEDULER_ON_SUNDAYS = 1 << 6
+};
+
+/** Representation of the actions to be executed for a scheduled event. */
+enum {
+	BT_MESH_SCHEDULER_TURN_OFF,
+	BT_MESH_SCHEDULER_TURN_ON,
+	BT_MESH_SCHEDULER_SCENE_RECALL,
+	BT_MESH_SCHEDULER_NO_ACTIONS = 0xf
+};
+
+/** Schedule Register entry. */
+struct bt_mesh_schedule_entry {
+	/** Two last digits of the scheduled year for the action, or
+	 * @ref BT_MESH_SCHEDULER_ANY_YEAR.
+	 */
+	uint16_t year : 7;
+	/** Scheduled month for the action. */
+	uint16_t month : 12;
+	/** Scheduled day of the month for the action. */
+	uint16_t day : 5;
+	/** Scheduled hour for the action. */
+	uint16_t hour : 5;
+	/** Scheduled minute for the action. */
+	uint16_t minute : 6;
+	/** Scheduled second for the action. */
+	uint16_t second : 6;
+	/** Schedule days of the week for the action. */
+	uint16_t day_of_week : 7;
+	/** Action to be performed at the scheduled time. */
+	uint16_t action : 4;
+	/** Transition time for this action. */
+	uint16_t transition_time : 8;
+	/** Scene number to be used for some actions. */
+	uint16_t scene_number : 16;
+};
+
+/** Scheduler Action status */
+struct bt_mesh_scheduler_action_status {
+	/** Enumerates (selects) a Schedule Register entry. */
+	uint8_t index;
+	/** Bit field defining an entry in the Schedule Register. */
+	struct bt_mesh_schedule_entry entry;
+};
+
+#define BT_MESH_SCHEDULER_OP_GET              BT_MESH_MODEL_OP_2(0x82, 0x49)
+#define BT_MESH_SCHEDULER_OP_STATUS           BT_MESH_MODEL_OP_2(0x82, 0x4A)
+#define BT_MESH_SCHEDULER_OP_ACTION_GET       BT_MESH_MODEL_OP_2(0x82, 0x48)
+#define BT_MESH_SCHEDULER_OP_ACTION_STATUS    BT_MESH_MODEL_OP_1(0x5F)
+#define BT_MESH_SCHEDULER_OP_ACTION_SET       BT_MESH_MODEL_OP_1(0x60)
+#define BT_MESH_SCHEDULER_OP_ACTION_SET_UNACK BT_MESH_MODEL_OP_1(0x61)
+
+#define BT_MESH_SCHEDULER_MSG_LEN_ACTION_GET 1
+#define BT_MESH_SCHEDULER_MSG_LEN_ACTION_SET 10
+#define BT_MESH_SCHEDULER_MSG_LEN_ACTION_STATUS 10
+#define BT_MESH_SCHEDULER_MSG_LEN_ACTION_STATUS_REDUCED 1
+#define BT_MESH_SCHEDULER_MSG_LEN_GET 0
+#define BT_MESH_SCHEDULER_MSG_LEN_STATUS 2
```

Comment:
Wrap these defines in 
`/** @cond INTERNAL_HIDDEN */`
`/** @endcond */`
like this: https://github.com/nrfconnect/sdk-nrf/blob/master/include/bluetooth/mesh/time.h#L89-L113